networking/firmware_protocol/src/lib.rs

```diff
@@ -90,17 +77,51 @@ impl SlimeString {
 }
 
 #[derive(Debug, PartialEq, DekuRead, DekuWrite)]
-#[deku(type = "u32")]
 #[deku(endian = "big")]
+pub struct Packet {
+	// TODO: This tag could really be dropped from the Rust side, but #[deku(temp)] is a bit wonky
+	tag: u32,
+	seq: u64,
+	#[deku(ctx = "*tag")]
+	ty_: PacketType,
```

Comment:
I'm not sure I understand this. Could you explain further and/or add some comments?

Things I'm confused on:
* `tag` appears to be the id of the `PacketType`. Why then do we store it in `Packet` and using it from the parent context, instead of keeping it the "normal" way and specifying the id in the `PacketType`?
* What is `seq`
* Why is there an `_` in `ty_`

Reply:
There is some inconsistency and confusing usage of "id" in existing SlimeVR code. Sometimes it refers to the enum variant of the packet, and sometimes it refers to the packet order number. Here I have chosen to use the name tag for the former (as in enum tag value) and seq for the latter (as in sequence number, common lingo in networking).

Sequence number is a value that is incremented for each packet. This way server can tell which packet is most recent one, as UDP traffic can appear out of order. This field is always common to all packets so it makes more sense to me to have it in a common struct, instead of writing it in each packet variant.

`PacketType` is the variant of each packet, and all logic just deals with them directly. `Packet` instead is the full, on the wire version of the packet, where the network stack has added the proper sequence numbering and the enum variant is encoded correctly. With bare enum it is not possible to describe `[variant_id] [sequence_num] [variant_data...]`

`ty_` was just brainfart on my part, could not choose between `ty` and `type_` :)

Reply:
I see, that all makes sense :)
I'll make a few small modifications when I'm ready to merge it shortly